site/assets/js/Pricing-0.4.0.js

```diff
@@ -0,0 +1,291 @@
+      price = planPricing.base[billingKey] + (planPricing.tier2[billingKey] * (increments - 1));
+    } else if (increments < 100) {
+      price = planPricing.base[billingKey] + (planPricing.tier2[billingKey] * 9) + (planPricing.tier3[billingKey] * (increments - 10));
+    } else {
+      price = planPricing.base[billingKey] + (planPricing.tier2[billingKey] * 9) + (planPricing.tier3[billingKey] * 90) + (planPricing.tier4[billingKey] * (increments - 100));
+    }
+
+    return price;
+  }
+
+  _changeStep() {
+    this._redraw();
+    if (this.step) {
+      this.url.searchParams.set('step', this.step);
+    }
+    if (this.hosting) {
+      this.url.searchParams.set('hosting', this.hosting);
+    }
+    if (this.plan) {
+      this.url.searchParams.set('plan', this.plan);
+    }
+  }
+
+  _handlePlanClick(event) {
+    // Let the download click bubble
+    const plan = event.currentTarget.dataset.plan;
+    if (this.hosting === 'self-hosting' && plan === 'Community') {
+      return;
+    }
+
+    event.stopPropagation();
+    event.preventDefault();
+    if (plan) {
+      this.plan = plan;
+    }
+    this._redraw();
+  }
+
+  _handleSliderChange() {
+    const mau = parseInt(this.monthlyActiveUserSlider.value);
+    let mauText = new Intl.NumberFormat('en').format(mau);
+    if (mau === 1000000) {
+      mauText += "+";
+    }
+    this.monthlyActiveUserValue.innerText = mauText;
+
+    const width = this.monthlyActiveUserSlider.offsetWidth;
+    const ratio = (mau - this.monthlyActiveUserSliderMin) / (this.monthlyActiveUserSliderMax - this.monthlyActiveUserSliderMin);
+    let left = ((ratio * (width - 30)) + 5);
+    if (left > (width / 2)) {
+      left -= 10;
+    }
+    if (left > (width - 50)) {
+      left -= 15;
+    }
+    this.monthlyActiveUserValue.style.left = left + 'px';
+    this._redraw();
+  }
+
+  _handleBillingIntervalChange() {
+    this.billingInterval = this.yearlyRadio.checked ? 'yearly' : 'monthly';
+    this._redraw();
+  }
+
+
+  _handleStateChange() {
+    this._loadState();
+    this._redraw();
+  }
+
+  _handleStepClick(event) {
+    event.stopPropagation();
+    event.preventDefault();
+    if (event.currentTarget.dataset.step === 'back') {
+      window.history.back();
+      return;
+    }
+
+    this.step = event.currentTarget.dataset.step;
+    if (event.currentTarget.dataset.hosting) {
+      this.hosting = event.currentTarget.dataset.hosting;
+    }
+    if (event.currentTarget.dataset.plan) {
+      this.plan = event.currentTarget.dataset.plan;
+    }
+    this._changeStep();
+    window.history.pushState({}, '', this.url);
+  }
+
+  _loadState() {
+    this.url = new URL(window.location);
+    this.step = this.url.searchParams.get('step');
+    this.hosting = this.url.searchParams.get('hosting');
+    this.plan = this.url.searchParams.get('plan');
+    if (this.step === null) {
+      this.step = 'start';
+    }
+  }
+
+  _drawPlanPrices() {
+    for(const plan in FusionAuthPriceCalculator.plans) {
```

Comment:
Minor formatting, I don't know if we have JavaScript formatting enabled through our formatting rules, but the `for` should be followed by a space. `for (const ...`